mono/metadata/handle.h

```diff
@@ -487,7 +487,7 @@ extern const MonoObjectHandle mono_null_value_handle;
 static inline void
 mono_handle_assign (MonoObjectHandleOut dest, MonoObjectHandle src)
 {
-	MONO_HANDLE_SUPPRESS (dest->__raw = (gpointer)(src ? MONO_HANDLE_RAW (src) : NULL));
+	MONO_HANDLE_SUPPRESS (dest->__raw = (MonoObject *)(src ? MONO_HANDLE_RAW (src) : NULL));
```

Comment:
Change is good but why was it failing?

Reply:
Because on windows it's not valid to assign a gpointer rvalue to a MonoObject * lvalue, according to VS's output.